internal/extensionapi/connection_route.go

```diff
@@ -39,10 +39,35 @@ func (s *ExtensionServer) generateWebUIBearerTokenURL(r *http.Request, workspace
 		return "", "", fmt.Errorf("user information not found in request headers")
 	}
 
-	// Generate webUI URL
-	webUIURL, err := s.generateBearerAuthWebUIURL(workspaceName, namespace)
+	// Get workspace and access strategy for signer configuration
+	ws, err := s.getWorkspace(namespace, workspaceName)
 	if err != nil {
-		return "", "", err
+		return "", "", fmt.Errorf("failed to get workspace: %w", err)
+	}
+
+	accessStrategy, err := s.getAccessStrategy(ws)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to get access strategy: %w", err)
+	}
```

Comment:
sanity-check: if I follow the logic, if the workspace has no access strategy, then it will result in a 500 / internal server error. Is that your intention?